libr/debug/p/native/linux/linux_debug.c

```diff
@@ -509,3 +514,1397 @@ RList *linux_desc_list (int pid) {
+	
+	if (!flags_str) {
+		eprintf ("VmFlags: not found\n");
+	}
+
+	if (!flags_str)
+		return true;	/* if we don't have VmFlags, just dump it. I'll fix it later on */
+	
+	while (*flags_str++ != ' ')
+		;
+	while(*flags_str++ == ' ')
+		;
+	flags_str--;
+
+	p = strtok (flags_str, " ");
+	while (p) {
+		eprintf ("dump_this_map: %s\n", p);
+		if (strncmp (p, "sh", 2) == 0) {
+			eprintf ("vmflags |= SH_FLAG\n");
+			vmflags |= SH_FLAG;
+		}
+		if (strncmp (p, "io", 2) == 0) {
+			eprintf ("vmflags |= IO_FLAG\n");
+			vmflags |= IO_FLAG;
+		}
+		if (strncmp (p, "ht", 2) == 0) {
+			eprintf ("vmflags |= HT_FLAG\n");
+			vmflags |= HT_FLAG;
+		}
+		if (strncmp (p, "dd", 2) == 0) {
+			eprintf ("vmflags |= DD_FLAG\n");
+			vmflags |= DD_FLAG;
+		}
+
+		p = strtok (NULL, " ");
+	}
+
+
+	if (!(vmflags & SH_FLAG))
+		vmflags |= PV_FLAG;
+	
+	eprintf ("vmflags: %u\n", vmflags);
+
+	/* first check for dd and io flags */
+	if (vmflags & DD_FLAG)
+		return false;
+	if (vmflags & IO_FLAG)
+		return false;
+
+	if (vmflags & HT_FLAG) {
+		if ((filter_flags & MAP_HUG_PRIV) && anonymous) {
+			eprintf ("filter_flags & MAP_HUG_PRIV\n");
+			return true;
+		}
+		if (filter_flags & MAP_HUG_SHR) {
+			eprintf ("filter_flags & MAP_HUG_SHR\n");
+			return true;
+		}
+	}
+
+	if (vmflags & SH_FLAG) {
+		if (filter_flags & MAP_ANON_SHR) {
+			eprintf ("filter_flags & MAP_ANON_SHR\n");
+			return true;
+		}
+		if (filter_flags & MAP_HUG_SHR) {
+			eprintf ("filter_flags & MAP_HUG_SHR\n");
+			return true;
+		}
+	}
+
+	if (vmflags & PV_FLAG) {	
+		if ((filter_flags & MAP_ANON_PRIV) && anonymous) {
+			eprintf ("filter_flags & MAP_ANON_PRIV\n");
+			return true;
+		}
+		if ((filter_flags & MAP_HUG_PRIV) && anonymous) {
+			eprintf ("filter_flags & MAP_HUG_PRIV\n");
+			return true;
+		}
+	}
+	
+	if (file_backed) {
+		if (filter_flags & MAP_FILE_PRIV) {
+			eprintf ("filter_flags & MAP_FILE_PRIV\n");
+			return true;
+		}
+		if (filter_flags & MAP_FILE_SHR) {
+			eprintf ("filter_flags & MAP_FILE_PRIV\n");
+			return true;
+		}
+	}
+
+	fseek (f, 0, SEEK_SET);
+
+	eprintf ("dump_this_map: nothing found, returning false\n");
+	return false;
+}
+	
+static linux_map_entry_t *linux_get_mapped_files(RDebug *dbg, ut8 filter_flags) {
+        linux_map_entry_t *me_head, *me_tail;
+        linux_map_entry_t *pmentry;
+        FILE *f;
+        ut64 start_addr;
+        ut64 end_addr;
+        ut64 offset;
+	ut64 inode;
+        ut8 flag_perm;
+        pid_t mypid;
+	bool is_anonymous;
+	bool file_backed;
+	bool ret;
+        char buff[4906];
+        char file[80];
+        char *name;
+        char *p;
+        char *end_line;
+        char *end_token;
+        FILE *f_smaps;
+        MAPS_FIELD maps_current;
+
+        me_head = me_tail = NULL;
+        name = NULL;
+        mypid = dbg->pid;
+
+        snprintf (file, sizeof (file), "/proc/%d/smaps", mypid);
+        f_smaps = fopen (file, "r");
+	if (!f_smaps)
+		return NULL;
+
+        memset (file, '\0', sizeof (file));
```

Comment:
I'm not sure you really need this call, because the next `snprintf` should also put the terminating char